app/models/order.rb

```diff
@@ -267,6 +268,28 @@ def can_transition
     return (required_process_checks - process_checklists).empty?
   end
 
+  def send_submission_pickup_email?
+    booking_type.eql?(BookingType.appointment) || order_transport&.pickup?
+  end
+
+  def send_submission_delivery_email?
+    booking_type == BookingType.online_order
+  end
+
+  def send_order_submission_email
+    return if created_by.nil? || !state.eql?("submitted")
+    sendgrid_instance = SendgridService.new(created_by)
+    begin
+      if send_submission_pickup_email?
```

Comment:
Do we need two methods here ?
They seem to do the same thing apart from using a different template ID, and they take the same argument.

maybe a simple 

```
sendgrid_instance.send_order_submission_email(self, type: type_of_the_order)
```

Really not a big deal, but the whole point of the `template_id ` mapping in the the sendgrid service, is to not have an `if/else` flow here. I think the method generation, although fun, is not bringing too much value